src/mpid/ch4/shm/xpmem/xpmem_impl.h

```diff
@@ -35,22 +36,92 @@
 #define XPMEM_PT2PT_DBG_PRINT(stmt, ...) do {} while (0)
 #endif
 
+/* MPIDI_XPMEM_seg_release check if the reference count of segment is 0.
+ * if 0, it decreases the segment map reference count, detach and release
+ * the segment. */
 #undef FUNCNAME
-#define FUNCNAME MPIDI_XPMEM_seg_regist_memrange
+#define FUNCNAME MPIDI_XPMEM_seg_release
+#undef FCNAME
+#define FCNAME MPL_QUOTE(FUNCNAME)
+MPL_STATIC_INLINE_PREFIX int MPIDI_XPMEM_seg_release(MPIDI_XPMEM_seg_t * seg)
+{
```

Comment:
Since the `release|create` functions are used as function pointers now, they do need to be static inline. Can you please move them to `globals.c` ? This can reduce the size of function inlining.